examples/asr/speech_to_text_finetune.py

```diff
@@ -0,0 +1,137 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This script can used to fine-tune a speech-to-text model of any instance type when users want to 
+fine-tune an existing model without changing its core architecture but may change the tokenizer. 
+One can mention the pretrained model in two ways: 
+1) `init_from_nemo_model` or 
+2) `init_from_pretrained_model` in the configuration.
+
+To update the model architecture in conjunction with other modifications, it is advisable to use the primary 'speech_to_text_rnnt/ctc_*.py' script.
+
+Note: To create a single script for all model types, we currently only support two types of 
+initializations:
+1) `init_from_nemo_model`, and
+2) `init_from_pretrained_model`,
+but not `init_from_ptl_ckpt`.
+
+To train with prior base model tokenizer keep `model.tokenizer.update_tokenizer` as false else
+make it true and provide tokenizer dir along with tokenizer type.
+
+To fine-tune the model, use the following commands:
+
+For initialization from a NEMO model:
+```sh
+python <NEMO_ROOT>/examples/asr/speech_to_text_finetune.py \
+    init_from_nemo_model=<path_to_nemo_model>
+```
+
+For initialization from a pretrained model:
+```sh
+python <NEMO_ROOT>/examples/asr/speech_to_text_finetune.py \
+    init_from_pretrained_model=<pretrained_model_name>
+```
+
+# Fine-Tune a Model
+
+For documentation on fine-tuning this model, please visit:
+https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/main/asr/configs.html#fine-tuning-configurations
+"""
+
+import copy
+
+import pytorch_lightning as pl
+from omegaconf import OmegaConf
+from pytorch_lightning.utilities import rank_zero_only
+
+from nemo.collections.asr.models import ASRModel
+from nemo.core.config import hydra_runner
+from nemo.utils import logging, model_utils
+from nemo.utils.exp_manager import exp_manager
+
+
+@hydra_runner(config_path="conf", config_name="speech_to_text_finetune")
+def main(cfg):
+    logging.info(f'Hydra config: {OmegaConf.to_yaml(cfg)}')
+
+    trainer = pl.Trainer(**cfg.trainer)
+    exp_manager(trainer, cfg.get("exp_manager", None))
+
+    if hasattr(cfg, 'init_from_ptl_ckpt') and cfg.init_from_ptl_ckpt is not None:
```

Comment:
Let's move stuff to different functions to keep things clean

Reply:
makes sense, to utils or modelPT?

Reply:
No, functions inside this script is fine probably. If later it turns out to be generic and useful we can pull it out into utils.